### PR TITLE
[v7r2] Yet more executor fixes

### PR DESCRIPTION
**Pull request by chrisburr**
453daa10a16c83446e8bff345b14d87ec33e9636 is the commit which actually fixes things, the other is just improving the logging.

BEGINRELEASENOTES

*Core
FIX: Race condition which causes multiple connectioons to be opened when the connection to an executor drops
CHANGE: Improve logging split between message/varmessage in execuctors

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Suggestions for titles of next PRs:
- Yet again more executor fixes
- Still executor fixes
- last executor fix
- you know what fix
- WTF exectorfixesaaaah

Jokes apart: OptimizationMindHandler would be a decent candidate for HTTPs, if only it would be duplicat-able. But given that we want to move all the executor machinery out, then we better refrain.

**Review by atsareg — APPROVED** [review]

**Comment by chaen** [comment]
executors were on purpose left out of all the HTTPs investigation exactly because we agreed (what was it ? 5 years ago now ? ) that it should go

File: src/DIRAC/Core/Utilities/ExecutorDispatcher.py [hunk 1]
```diff
@@ -836,18 +832,12 @@ def __sendTaskToExecutor(self, eId, eTypes=False, checkIdle=False):
         return S_OK(taskId)
 
     def __msgTaskToExecutor(self, taskId, eId, eType):
-        try:
-            self.__tasks[taskId].sendTime = time.time()
-        except KeyError:
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/ExecutorDispatcher.py:841]: Is it really OK to drop this check?

**Comment by fstagni** [comment]
Rebase it please, to see the tests passing.

File: src/DIRAC/Core/Utilities/ExecutorDispatcher.py (hunk 1 above)
Comment (chrisburr) [reply src/DIRAC/Core/Utilities/ExecutorDispatcher.py:841]: It's only re-raising the exception and all expceptions are caught by `__sendTaskToExecutor` a few lines higher.

**Comment by chrisburr** [comment]
> Rebase it please, to see the tests passing.

Closing an re-opening should have the same effect (as GitHub actions is configured to test the merge commit assuming this was just merged).

**Comment by chrisburr** [comment]
> Jokes apart: OptimizationMindHandler would be a decent candidate for HTTPs, if only it would be duplicat-able. But given that we want to move all the executor machinery out, then we better refrain.

The last thing it needs is more race conditions... 😄

**Pull request merged**